### ISSUE TITLE
Unable to compare types anymore

### ISSUE THREAD

**Issue by vberlier** [microsoft/pyright#1244]
Hello! Looks like in the latest release pyright thinks that it's not okay to compare types for equality:

```python
type('foo') == type('foo')
```

```
Operator "==" not supported for types "Type[Literal['foo']]" and "Type[Literal['foo']]"
  Operator "==" not supported for types "Type[Literal['foo']]" and "Type[Literal['foo']]"
```

**Comment by erictraut** [comment]
Thanks for the bug report. There was a latent bug that was exposed by a recent change. I've fixed the bug and added unit tests to ensure that this is covered in the test suite. The fix will be in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.93, which I just published. It will also be included in the next release of Pylance.
